modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PageMemoryIndexes.java

```diff
@@ -105,8 +109,17 @@ void performRecovery(
                 StorageIndexDescriptor indexDescriptor = indexDescriptorSupplier.get(indexId);
 
                 if (indexDescriptor == null) {
-                    // TODO: IGNITE-21671 destroy the index if it can't be found in the Catalog.
-                    continue;
+                    // Index has not been found in the Catalog, we can treat it as destroyed and remove the storage.
+                    runConsistently.accept(locker -> {
+                        destroyIndexOnRecovery(indexMeta, indexStorageFactory, indexMetaTree)
+                                .whenComplete((v, e) -> {
+                                    if (e != null) {
+                                        LOG.error("Unable to destroy existing index that has been removed from the Catalog", e);
```

Comment:
Let's indicate which index could not be deleted.

Reply:
Fixed